libs/ui/src/components/CommentForm/CommentForm.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  Avatar,
+  Button,
+  HStack,
+  IconButton,
+  Stack,
+  Text,
+  Textarea,
+  useBreakpointValue,
+  VStack,
+} from '@chakra-ui/react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useTranslation } from 'next-i18next'
+import { useForm } from 'react-hook-form'
+import { FiArrowRight } from 'react-icons/fi'
+import * as yup from 'yup'
+
+import { FormItem } from '../FormItem'
+import { CommentFormProps } from './types'
+
+import { mutation } from '~lib'
+import { toastMessage } from '~utils'
+
+const userSchema = (t: TFunction<'translation', undefined>) =>
+  yup.object({
+    content: yup.string().required(t`comment-form.content.required`),
+  })
+
+const publicSchema = (t: TFunction<'translation', undefined>) =>
+  yup.object({
+    name: yup.string().required(t`comment-form.name.required`),
+    email: yup
+      .string()
+      .email(t`comment-form.email.invalid`)
+      .required(t`comment-form.email.required`),
+    content: yup.string().required(t`comment-form.content.required`),
+  })
+
+export const CommentForm: React.FC<CommentFormProps> = ({
+  auth,
+  onSendForm,
+}) => {
+  const { t } = useTranslation()
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+  } = useForm({

```

Comment:
```suggestion
  } = useForm<CommentFormFieldValues>({
```

It's better to give field values types to useForm